app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorEffectHandler.kt

```diff
@@ -28,8 +30,18 @@ class SyncIndicatorEffectHandler @AssistedInject constructor(
           .subtypeEffectHandler<SyncIndicatorEffect, SyncIndicatorEvent>()
           .addTransformer(FetchLastSyncedStatus::class.java, fetchLastSyncedState())
           .addTransformer(FetchDataForSyncIndicatorState::class.java, fetchDataForSyncIndicatorState())
+          .addTransformer(StartSyncedStateTimer::class.java, startTimer())
           .build()
 
+  private fun startTimer(): ObservableTransformer<StartSyncedStateTimer, SyncIndicatorEvent> {
+    return ObservableTransformer { effect ->
+      effect
+          .flatMap { Observable.interval(it.intervalAmount, it.timeUnit) }
```

Comment:
Should this be a `switchMap` instead? If we receive a new emission from upstream, we don't want two simultaneous ticks to be happening.

Reply:
In addition, would we want to cancel this in some way if the last synced state preference changes? I have no idea what that would look like, though.

Reply:
This timer will only get started in `Synced` state. This state itself won't be emitted simultaneously. I don't understand how will this happen?

Reply:
`.interval` will emit forever and since we are using a `.flatMap`, the previous emission won't be cancelled when upstream emits a new effect.

It is possible for the sync state to go from synced to pending and back to synced, which would cause a new effect to be emitted from upstream and a new `interval` subscription to be setup.

Reply:
That makes sense. I will make this change.